### ISSUE TITLE
Setting custom colors for `<Seperator>`s

### ISSUE THREAD

**Issue by DusterTheFirst** [Purii/react-native-tableview-simple#114]
Hello! I want to start out by thanking you for all the work you have put into this project and to say just how useful this has been with my app. I have one issue/question to file though.

I cannot find a way to change the left most padding section of the `<Seperator>`. Currently my settings screen (in dark theme) looks like this:
<img width="320" alt="noselect" src="https://user-images.githubusercontent.com/14093962/73144038-fccb8d00-406e-11ea-9249-c6b4c786ee65.png"/>
with those weird white sections on the left. Looking through the docs and the properties layed out in the typescript declaration file, I cannot find any way to change it. As well as the padding on the left, when a cell is selected, the whole `<Seperator>` changes to white:
![select](https://user-images.githubusercontent.com/14093962/73144053-356b6680-406f-11ea-9c2f-bba22ce53175.png)

For reference, here is how it looks in light theme:

![IMG_0307](https://user-images.githubusercontent.com/14093962/73144088-a27efc00-406f-11ea-84f2-c176cb8057cf.png)
![IMG_0306](https://user-images.githubusercontent.com/14093962/73144089-a3b02900-406f-11ea-909b-eb781402536c.png)

**Comment by Purii** [comment]
Hi @DusterTheFirst! Thanks for those kind words. I'm glad that you find that component useful for your work :)

How do you set the `backgroundColor` of the `Cells`?

```jsx
// Did you try that?
<Cell backgroundColor="darkColor" />
```

The `backgroundColor` of a `Separator` is derived by the surrounding `Cells`: https://github.com/Purii/react-native-tableview-simple/blob/master/src/components/Section.js#L124

The reason for that is the onPress- / highlighting-behavior, where the `Separator` is not hidden but expanded to full width with the highlight color. Otherwise the components would jump.

Should add a dark theme for the component 🤔

**Comment by DusterTheFirst** [comment]
@Purii I have the background color set on the cell
![image](https://user-images.githubusercontent.com/14093962/73196317-66927800-40fd-11ea-868c-ee61808a1a16.png)

**Comment by Purii** [comment]
Strange. Unfortunately I cannot reconstruct that issue.
Could you create an expo snack where it occurs?
Thanks!

**Comment by DusterTheFirst** [comment]
@Purii here is an example: https://snack.expo.io/@dusterthefirst/06c77c
I think the main problem has to do with the fact that the components are passed the backgroundColor through a HOC.

**Comment by Purii** [comment]
You're right. I'll figure something out. Probably a recursion to get the correct children...

I'm also open for suggestions or a PR if you find a solution 👍

**Comment by DusterTheFirst** [comment]
@Purii i don’t have the time currently but I will look into solutions and probably submit a PR. Am idea I have is to, as you said, recurse through the components passed and their children until you hit one that is a cell (or array of cells) and deal with them like you are currently.  Another, probably quicker solution could also be to add a cellBackgroundColor prop to Section so that it can be used if there is none specified for the specific cell.

**Comment by Purii** [comment]
Hi @DusterTheFirst !

What do you think about this clean solution:

```
const cellPropsDefault = {
  cellStyle: 'Basic',
  title: 'Basic Custom',
  backgroundColor: 'grey',
};

...

<Cell onPress={console.log} {...cellPropsDefault} />
```

**Comment by Purii** [comment]
*Just reopen the issue, if it is still not fixed.*

**Comment by CosmicClownTyler** [comment]
I see this is 4 years old but I figured out the specifics of this issue. I raised a new issue [here](https://github.com/Purii/react-native-tableview-simple/issues/784). The short version is that if you are creating a custom cell component that is returning a cell with specific props, the background color defined in the custom component is used to color the cell but _not_ the inset space. The background color passed when using the custom component is used for the inset, even if no background color is passed. More details can be found in the issue I created.
